### PR TITLE
2553: Add Accessibility Role to components

### PR DESCRIPTION
**Pull request by bahaaTuffaha**
### Short description

For better accessibility: the role should be set.

### Proposed changes

- Replaced `accessibilityRole` with `role`
- `ThemedSearchBar.tsx` has search bar with `role ="searchbox"` instead of  `role ="search"`
- Forcing role for `Pressable from './base/Pressable'`

### Side effects

None.

### Resolved issues

Fixes: #2553 

---

### CODE REVIEW

**Comment by LeandraH** [comment]
Is this ready for review or still a draft?

**Comment by bahaaTuffaha** [comment]
> Is this ready for review or still a draft?

The replacing part is done but I need to look for any other component that needs `Role` tag.
When it's ready I will click **Ready for review**

File: native/src/components/NavigationTile.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ type NavigationTileProps = {
 }
 
 const NavigationTile = ({ tile, width }: NavigationTileProps): ReactElement => (
-  <StyledPressable onPress={tile.onTilePress} width={width}>
+  <StyledPressable role='button' onPress={tile.onTilePress} width={width}>
```
Comment (LeandraH) [inline native/src/components/NavigationTile.tsx:53]: I think this is also a link, when tapping on a NavigationTile, you are sent to the route that you just tapped on.

File: native/src/components/Tile.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ const Tile = ({ onTilePress, tile, resourceCache }: TileProps): ReactElement =>
     tile.isExternalUrl ? openExternalUrl(tile.path, showSnackbar).catch(reportError) : onTilePress(tile)
 
   return (
-    <TileContainer onPress={openTile}>
+    <TileContainer onPress={openTile} role='button'>
```
Comment (LeandraH) [inline native/src/components/Tile.tsx:43]: This also looks like a link to me

**Review by LeandraH — APPROVED** [review]
Nice, thank you! Not tested

**Review by steffenkleinle — COMMENTED** [review]
Thanks for this PR!

I just found https://www.npmjs.com/package/eslint-plugin-react-native-a11y and I think we should employ it to avoid forgetting this in the future. I just installed it and it seems like the role is still missing in 5 cases (at least I am getting a linting error). Could you perhaps check out https://github.com/digitalfabrik/integreat-app/pull/2913 and merge this if you think this makes sense?

Sadly these rules seem to not work together with styled-components/custom components. Therefore it would be nice to add a required property `role` to `base/Pressable` to also fix future problems. Thank you!

**Comment by bahaaTuffaha** [comment]
> Thanks for this PR!
> 
> I just found https://www.npmjs.com/package/eslint-plugin-react-native-a11y and I think we should employ it to avoid forgetting this in the future. I just installed it and it seems like the role is still missing in 5 cases (at least I am getting a linting error). Could you perhaps check out #2913 and merge this if you think this makes sense?
> 
> Sadly these rules seem to not work together with styled-components/custom components. Therefore it would be nice to add a required property `role` to `base/Pressable` to also fix future problems. Thank you!

I decided not to proceed with installing "eslint-plugin-react-native-a11y" because styled-components didn't work properly with it. also we rarely used vanilla React Native components without styled-components.

File: native/src/components/base/SettingsSwitch.tsx [hunk 3]
```diff
@@ -11,14 +11,15 @@ const SettingsSwitch = ({ value, onPress }: SettingsSwitchProps): ReactElement =
   const theme = useTheme()
   return (
     <Switch
+      testID='SettingsSwitch'
```
Comment (steffenkleinle) [inline native/src/components/base/SettingsSwitch.tsx:14]:
```suggestion
```

File: native/src/components/SettingItem.tsx [hunk 4]
```diff
@@ -47,18 +47,18 @@ type SettingItemProps = {
...
 }
 
 const SettingItem = (props: SettingItemProps): ReactElement => {
-  const { title, description, onPress, value, hasBadge, hasSwitch, bigTitle, accessibilityRole } = props
+  const { title, description, onPress, value, hasBadge, hasSwitch, bigTitle, role } = props
   const { t } = useTranslation('settings')
 
   return (
-    <Pressable onPress={onPress} accessibilityRole={accessibilityRole} accessible={false}>
+    <Pressable onPress={onPress} role={role || 'none'} accessible={false}>
```
Comment (steffenkleinle) [inline native/src/components/SettingItem.tsx:61]:
```suggestion
    <Pressable onPress={onPress} role={role ?? 'none'} accessible={false}>
```

**Review by steffenkleinle — APPROVED** [review]
Nice work, thanks!

**Pull request merged**